reporter/parca_reporter.go

```diff
@@ -244,7 +241,9 @@ func (r *ParcaReporter) ExecutableMetadata(ctx context.Context,
 	}
 
 	// Always attempt to upload, the uploader is responsible for deduplication.
-	r.uploader.Upload(ctx, fileID, buildID, open)
+	// !! FIXME !! Apparently uploading stuff in this function is a "really bad idea";
+	// see https://github.com/open-telemetry/opentelemetry-ebpf-profiler/pull/126#pullrequestreview-2248554710
```

Comment:
The recommendation is to specifically not *block* this Go routine, but instead open the file and pass the file handle to an asynchronous mechanism. Apart from immediately opening the file, that's exactly how our mechanism works, so we can remove the context here without a problem.